state.go

```diff
@@ -357,8 +357,14 @@ func (uv *UpdateVerifyState) Handle(ctx *StateContext, c Controller) (State, boo
 	}
 
 	if has {
-		if uv.update.ArtifactName() == c.GetCurrentArtifactName() {
-			log.Infof("successfully running with new image %v", c.GetCurrentArtifactName())
+		artifactName, err := c.GetCurrentArtifactName()
+		if err != nil {
+			log.Errorf("Cannot determine current artifact. Update will note continue: %v : %v", defaultDeviceTypeFile, err)
+			me := NewFatalError(errors.Wrapf(err, "Cannot determine current artifact. Update will not continue: %v : %v", defaultDeviceTypeFile, err))
```

Comment:
I suggest changing both of these to:

```
log.Errorf("Cannot determine name of new artifact. Update will not continue: %v : %v", defaultDeviceTypeFile, err)
me := NewFatalError(errors.Wrapf(err, "Cannot determine name of new artifact. Update will not continue: %v : %v", defaultDeviceTypeFile, err))
```

Just to emphasize that the issue is in the *new* artifact, since this is the only place where you actually have a shot at fixing the issue.